File: typescript/test/FakeCatalog.ts
import {Product} from "../src/model/Product"
import {SupermarketCatalog} from "../src/model/SupermarketCatalog"

export class FakeCatalog implements SupermarketCatalog {
    private products: {[key: string]: Product} = {};
    private prices: {[key: string]: number} = {};

    public addProduct(product: Product, price: number): void {
        this.products[product.name] = product;
        this.prices[product.name] = price;
    }

    public getUnitPrice(p: Product): number {
        return this.prices[p.name];
    }
}

File: cpp/test/FakeCatalog.cpp
//
// Created by sdargo on 2/4/19.
//

#include "FakeCatalog.h"

void FakeCatalog::addProduct(const Product& product, double price) {
    products[product.getName()] = product;
    prices[product.getName()] = price;
}

double FakeCatalog::getUnitPrice(const Product& product) {
    return prices[product.getName()];
}
